demo-orgs: Use banners for org settings demo organization warning.

laurynmm opened the pull request:
Fixes #34447 (all other points in issue have been completed in other pull requests):
> **Top of org settings banner**
> Use the same banner as in the navbar (color, buttons, etc.).
>
> Text: remove all links
> Buttons:
>  - Learn more
>      - low emphasis
>      - link to /help/demo-organizations
>  - Convert
>      - medium emphasis
>      - opens conversion modal

**Notes**:
- Migrates the demo organization warning that's shown on all the tabs for organization settings to use the shared banner code, so the color variables for the demo organization warning are no longer needed.
- There are a few differences in the `Banner` used in org settings and the `AlertBanner` used in the navbar alerts, so there's some duplication between the two modules.
- With a narrow browser window the banner's buttons take up a lot of space. As there is other ongoing work on implementing banners in the settings overlay, I wasn't sure if that styling's being addressed in those changes and therefore would apply to these demo organization warning banners as well.
- Once the organization is 7 days from deletion, the banner intent switches from info to danger, which is the same for the navbar alert banner.

---

**Screenshots and screen captures:**

### Org settings banner for owners:
| Description | Screenshot |
| --- | --- |
| Before (link opened the conversion modal) | ![Screenshot from 2025-04-30 17-18-54](https://github.com/user-attachments/assets/eeb3fd78-bed5-48f4-96a6-2aabf9837623)
| After | ![Screenshot from 2025-04-30 17-17-21](https://github.com/user-attachments/assets/297d9d4d-4816-42cc-a700-cbc19fd4606b)

### Org settings banner for non-owners:
| Description | Screenshot |
| --- | --- |
| Before (link opened the help center article) | ![Screenshot from 2025-04-30 17-18-32](https://github.com/user-attachments/assets/28fb2ce5-6d34-4efa-a985-ef4327d9f39c)
| After | ![Screenshot from 2025-04-30 17-17-50](https://github.com/user-attachments/assets/e30fc3a0-6101-4e20-b0dc-6d46f82b529b)

### Narrow browser window - owners case:
| Description | Screenshot |
| --- | --- |
| Before (link opened the conversion modal) | ![Screenshot from 2025-05-01 15-45-51](https://github.com/user-attachments/assets/a32d249b-3c9e-410f-9c4a-171a718c8f86)
| After | ![Screenshot from 2025-05-06 12-57-34](https://github.com/user-attachments/assets/08dbb5b6-2116-4748-9bc4-e52cc2e524f0)



---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/src/demo_organizations_ui.ts:
```diff
@@ -108,19 +144,11 @@ export function do_convert_demo_organization(): void {
-                $(this).trigger("click");
-            }
-        },
-    );
```
Is this enter-acts-like-click behavior present elsewhere? I don't think it's very important -- it's kinda hard to tab to the component -- so it's mostly a question of whether our banner component already does this.

alya commented:
At some point on narrow enough screens, the buttons show [move below the text](https://www.figma.com/design/msWyAJ8cnMHgOMPxi7BUvA/Zulip-Web-UI-kit?node-id=694-4097&t=PgfXVAwwX4aPSFSd-0). @sayamsamal I don't know why that's not happening here?

laurynmm commented:
**Updates**:
- Addressed the button placement and banner text alignment in narrow windows (see updated screenshots in original PR comment).
- Added a prep commit to add `e.preventDefault` and `e.stopPropagation` to the event listeners for opening the convert demo organization modal. I didn't add this to the even listener for opening the help center in a separate browser window.

laurynmm commented on web/src/demo_organizations_ui.ts:
```diff
@@ -108,19 +144,11 @@ export function do_convert_demo_organization(): void {
 export function handle_demo_organization_conversion(): void {
-    $(".convert-demo-organization-button").on("click", () => {
+    $(".demo-organization-warning").on("click", ".convert-demo-organization", () => {
         do_convert_demo_organization();
```
Yeah, I'd been using `web/src/navbar_alerts.ts` as a bit of a guide for these updates and the related changes in #34518. Most of the click handlers there for the various banner buttons don't specifically use `e.preventDefault`.

Added a prep commit for the event listeners that call `do_convert_demo_organization`.

sayamsamal replied:
Yeah, most of the click handlers are delegated to the parent container and hence we shouldn't be facing any event bubbling issues, but yeah, we may as well use it for best practices.

alya commented:
@shuklamaneesh23 Are we handling buttons in narrow windows correctly in other settings banners?

shuklamaneesh23 commented:
> @shuklamaneesh23 Are we handling buttons in narrow windows correctly in other settings banners?

@alya , if you are talking about the setting banners that i'm handling in # , then i guess the current situation is not handling the buttons correctly.

| At large screens | Narrow screens |
|---|---|
| <img width="682" alt="Screenshot 2025-05-02 at 1 27 26 PM" src="https://github.com/user-attachments/assets/2947f9d8-eaec-44b6-bf47-472a2b76e49c" /> | <img width="464" alt="Screenshot 2025-05-02 at 1 27 20 PM" src="https://github.com/user-attachments/assets/072d85a5-5f31-4843-b650-f51c3ab9d69c" /> |

I think we need to rework on the banner design which is `web/templates/components/banner.hbs`  to handle the narrow windows correctly rather than working separately for each banner at different locations.

sayamsamal commented:
> At some point on narrow enough screens, the buttons show [move below the text](https://www.figma.com/design/msWyAJ8cnMHgOMPxi7BUvA/Zulip-Web-UI-kit?node-id=694-4097&t=PgfXVAwwX4aPSFSd-0). @sayamsamal I don't know why that's not happening here?

I think it was just the missing `banner-wrapper` class in the banner's parent div required for the container queries to work, @shuklamaneesh23 that's probably the issue you are facing as well.

sayamsamal commented on web/styles/banners.css:
```diff
@@ -61,16 +61,20 @@
         auto 0 auto minmax(0, 1fr) minmax(0, auto) var(
             --banner-horizontal-padding
         );
+    place-items: start center;
```
Are we sure about centering the banner buttons in this particular banner on narrower widths? I think we are going with the normal left alignment with all the other settings banners, so wouldn't this be a odd one then?

laurynmm replied:
The Figma link that @alya referenced [here](https://github.com/zulip/zulip/pull/34527#issuecomment-2843743886) only shows the navbar banner design. Is there a CZO discussion or GitHub discussion about the design for the banners in the settings overlays? Or a better Figma link?

laurynmm replied:
### Narrowed browser window
| Centered button | Left-aligned button |
| --- | --- |
| ![Screenshot from 2025-05-05 16-26-16](https://github.com/user-attachments/assets/97f41954-c9c4-4020-935b-10eed300dc17) | ![Screenshot from 2025-05-05 16-26-05](https://github.com/user-attachments/assets/2331a63e-298e-44de-811f-4b6af3195558)

sayamsamal replied:
You'll find the Figma designs for the base banners on the left of the same link (https://www.figma.com/design/msWyAJ8cnMHgOMPxi7BUvA/Zulip-Web-UI-kit?node-id=510-2932&t=730QgrQjlUr81jq3-0)

sayamsamal replied:
There probably isn't a specific discussion about the banners for the settings overlay, but I'm not too sure. I just remember this detail that we wanted the navbar banner elements to be center aligned because they span across the viewport and that could be pretty wide — So the label and the action buttons would be too wide apart. Rest of the banners wouldn't face this issue as they are mostly constricted to a sane max width and hence the normal banner layout.

sayamsamal replied:
The above screenshot of the left-aligned version looks weird because the text in it is probably still center aligned and not left aligned as expected.

sayamsamal replied:
@alya You would be the better person to make the decision on whether to have the normal banner layout (left-aligned) or the navbar banner style layout (center aligned) for the banners in the settings overlay.

alya replied:
Left-aligned seems fine (with left-aligned text, not as in the screenshots above). @terpimost does that sound right?

laurynmm replied:
I've updated the changes here so that the banner text and buttons remain left-aligned. See updated screenshots [below](https://github.com/zulip/zulip/pull/34527#issuecomment-2854196732).

shuklamaneesh23 commented:
> > At some point on narrow enough screens, the buttons show [move below the text](https://www.figma.com/design/msWyAJ8cnMHgOMPxi7BUvA/Zulip-Web-UI-kit?node-id=694-4097&t=PgfXVAwwX4aPSFSd-0). @sayamsamal I don't know why that's not happening here?
> 
> I think it was just the missing `banner-wrapper` class in the banner's parent div required for the container queries to work, @shuklamaneesh23 that's probably the issue you are facing as well.

I tried putting the banner-wrapper class in the parent div and the alignment of the banner is getting worse. I'm trying to debug why it's happening here but couldn't find anything leading.

sayamsamal commented on web/src/demo_organizations_ui.ts:
```diff
@@ -26,13 +28,48 @@ export function get_demo_organization_deadline_days_remaining(): number {
+    const demo_organization_warning_banner: banners.Banner = {
+        intent: days_remaining <= 7 ? "danger" : "info",
+        label: new Handlebars.SafeString(
+            $t_html(
+                {
+                    defaultMessage:
+                        "This demo organization will be automatically deleted in {days_remaining} days, unless it's converted into a permanent organization.",
+                },
+                {
+                    days_remaining,
+                },
+            ),
+        ),
```
Nothing critical, but we don't need `$t_html()` over here and can just do away with `$t()` translation tag since no html is involved here anymore.

laurynmm replied:
Thanks for catching that! Added a prep commit to clean up the nabar alert for demo organizations too.

laurynmm commented:
**Updates**:
- Added a prep commit that updates the demo organization navbar alert to use the `$t()` translation tag since no html is involved for these banners anymore. And uses that for the organization settings banner as well.
- The buttons for the organization settings banner are still centered for narrow browser windows; see [this comment](https://github.com/zulip/zulip/pull/34527#discussion_r2073555917) for screenshots of both options.

laurynmm commented:
**Updates**:
- When browser window is narrowed, the banner text and buttons stay left-aligned, and the buttons move to under the banner text. See updated screenshots below, and the narrowed example in the original PR comment above.

---

**Updated  screenshots**:

**Owner's account**
| Description | Screenshot |
| --- | --- |
| Full width | ![Screenshot from 2025-05-06 13-04-21](https://github.com/user-attachments/assets/512de316-d869-4500-b7dd-655b688df90e)
| Slightly narrowed | ![Screenshot from 2025-05-06 12-56-18](https://github.com/user-attachments/assets/79912f96-c345-4d97-bc2c-143e59a01645)
| Narrowed, triggers button move | ![Screenshot from 2025-05-06 12-55-50](https://github.com/user-attachments/assets/d508a707-b92f-4ef4-90b4-18789d88210d)
| Narrowed, settings tabs hidden | ![Screenshot from 2025-05-06 12-57-34](https://github.com/user-attachments/assets/b20876e3-6fc8-46ae-8021-daf548322e83)
| Dev tools mobile view | ![Screenshot from 2025-05-06 13-16-00](https://github.com/user-attachments/assets/0e785a2b-2b12-42f3-a1b6-9891d3a634bf)

**Other users**
| Description | Screenshot |
| --- | --- |
| Full width | ![Screenshot from 2025-05-06 13-04-46](https://github.com/user-attachments/assets/f9aa5aea-024c-4a3e-a94b-8a88a8cf46b5)
| Slightly narrowed | ![Screenshot from 2025-05-06 12-56-33](https://github.com/user-attachments/assets/3955f105-38b5-4aab-a554-32cefbcfa61e)
| Narrowed, triggers button move | ![Screenshot from 2025-05-06 12-56-49](https://github.com/user-attachments/assets/8808dcb1-03d9-42e9-a129-68504d286d40)
| Narrowed, settings tabs hidden | ![Screenshot from 2025-05-06 12-57-05](https://github.com/user-attachments/assets/d87cd796-0cb0-464c-8cec-28b76ad8087a)
| Dev tools mobile view | ![Screenshot from 2025-05-06 13-15-16](https://github.com/user-attachments/assets/01e007ea-9e72-409c-a6cc-c881afd7edcf)

alya commented:
The screenshots look good to me!

alya commented:
@timabbott I think this is ready for your review.

timabbott commented:
Merged, thanks @laurynmm!
